dhall-lsp-server/src/Dhall/LSP/Backend/Diagnostics.hs

```diff
@@ -119,15 +120,23 @@ positionToMegaparsec (line, col) = Megaparsec.SourcePos ""
                                      (Megaparsec.mkPos $ max 0 line + 1)
                                      (Megaparsec.mkPos $ max 0 col + 1)
 
+addRelativePosition :: Position -> Position -> Position
+addRelativePosition (x1, y1) (0, dy2) = (x1, y1 + dy2)
+addRelativePosition (x1, _) (dx2, y2) = (x1 + dx2, y2)
+
+-- addRelativePosition pos (relativePosition pos pos') = pos'
```

Comment:
If you enable `doctest` for the `dhall-lsp-server` package then you could turn this into a checked property test:

```haskell
-- | prop> addRelativePosition pos (relativePosition pos pos') == pos'
```